Draft spec: Operator incentives (#44)
waku-org/specs: informational/operator-incentives.md

---
title: WAKU2-OPERATOR-INCENTIVES
name: Waku Operator Incentives
category: Standards Track
tags: [waku/core-protocol]
editor: Alvaro R. <alrevuelta@status.im>
contributors:
---

## Abstract

> This draft spec proposes how to economically incentivise node operators to participate in waku, by distributing the rewards paid by users when registering their RLN memberships.

Reviewer: ```suggestion
This draft spec proposes how to economically incentivise node operators to participate in Waku RLN Relay, by distributing the rewards paid by users when registering their RLN memberships.
```
no?

## Syntax

The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).

## Background

This draft spec comes from [here](https://github.com/waku-org/research/issues/101).

The waku ecosystem contains two main economic agents:
* Users: Make use of the network to send messages. They hold a RLN membership with some associated cost. They are the consumers.
* Operators: Run the network infra, providing services to users in the network. They are the producers.

This proposal connects them together by organising in a decentralized way how users and operators can interact with each other via a peer-to-peer network, settling
all decisions in a third-party blockchain, making use of smart contracts.

Definitions:
* Operator: Entity running `nwaku` that has deposited a stake and is eligible for root voting.
* Stake: Collateral posted by each operator, entitling it to participate in consensus.
* Slash: Penalty applied to a node operator, subtracted from its stake.

> * node vs operator. uno tiene stake el otro no.

Reviewer: ```suggestion
* node vs operator. one holds a stake, the other not.
```
Not sure the Spanish here was intended
